fsspec/spec.py

```diff
@@ -146,7 +147,23 @@ def __init__(self, *args, **storage_options):
         self._intrans = False
         self._transaction = None
         self._invalidated_caches_in_transaction = []
-        self.dircache = DirCache(**storage_options)
+        if not listings_cache_options:
+            listings_cache_options = {}
+        elif listings_cache_options is True:
+            listings_cache_options = {"cache_type": CacheType.MEMORY, "expiry_time": None}
+        else:
+            listings_cache_options = {"expiry_time": None} | listings_cache_options
+            cache_type = listings_cache_options.get("cache_type")
+            if cache_type:
+                if isinstance(cache_type, str):
+                    try:
+                        cache_type = CacheType[cache_type.upper()]
+                    except KeyError as e:
+                        raise ValueError(
+                            f"Cache type must be one of {', '.join(ct.name.lower() for ct in CacheType)}") from e
+                listings_cache_options["cache_type"] = cache_type
```

Comment:
I think there's too much going on here!

Reply:
I think it's fine! It could be a `norm_cache_options` util functions to pull it out of here.